src/ui/public/timepicker/__tests__/toggle.js

```diff
@@ -24,28 +25,60 @@ describe('kbnGlobalTimepicker', function () {
     expect($el.attr('data-test-subj')).to.be('globalTimepicker');
   });
 
-  it('sets data-shared-timefilter to true when auto-refresh selector is enabled', function () {
+  it('sets data-shared-timefilter-* using the timefilter when auto-refresh selector is enabled', function () {
+    const minString = '2000-01-01T00:00:00Z';
+    const maxString = '2001-01-01T00:00:00Z';
+    const bounds = {
+      min: moment(minString),
+      max: moment(maxString),
+    };
     scope.timefilter = {
       isAutoRefreshSelectorEnabled: true,
-      isTimeRangeSelectorEnabled: false
+      isTimeRangeSelectorEnabled: false,
+      getBounds: () => bounds
     };
+
     const $el = compile();
-    expect($el.attr('data-shared-timefilter')).to.eql('true');
+
+    expect($el.attr('data-shared-timefilter-from')).to.eql(minString);
+    expect($el.attr('data-shared-timefilter-to')).to.eql(maxString);
   });
-  it('sets data-shared-timefilter to true when time range selector is enabled', function () {
+
+  it('sets data-shared-timefilter-* using the timefilter when time range selector is enabled', function () {
+    const minString = '2000-01-01T00:00:00Z';
+    const maxString = '2001-01-01T00:00:00Z';
+    const bounds = {
+      min: moment(minString),
+      max: moment(maxString),
+    };
     scope.timefilter = {
       isAutoRefreshSelectorEnabled: false,
-      isTimeRangeSelectorEnabled: true
+      isTimeRangeSelectorEnabled: true,
+      getBounds: () => bounds
     };
+
     const $el = compile();
-    expect($el.attr('data-shared-timefilter')).to.eql('true');
+
+    expect($el.attr('data-shared-timefilter-from')).to.eql(minString);
+    expect($el.attr('data-shared-timefilter-to')).to.eql(maxString);
   });
-  it(`sets data-shared-timefilter to false when auto-refresh and time range selectors are both disabled`, function () {
+
+  it(`sets data-shared-timefilter-* to null when auto-refresh and time range selectors are both disabled`, function () {
+    const minString = '2000-01-01T00:00:00Z';
+    const maxString = '2001-01-01T00:00:00Z';
+    const bounds = {
+      min: moment(minString),
+      max: moment(maxString),
+    };
     scope.timefilter = {
       isAutoRefreshSelectorEnabled: false,
-      isTimeRangeSelectorEnabled: false
+      isTimeRangeSelectorEnabled: false,
+      getBounds: () => bounds
     };
+
     const $el = compile();
-    expect($el.attr('data-shared-timefilter')).to.eql('false');
+
+    expect($el.attr('data-shared-timefilter-from')).to.eql('');
```

Comment:
Contrary to the test description, this isn't testing for `null`. The code that uses this is checking for a truthy value... maybe the test should do the same?

Reply:
When there's an attribute added without a value, like this:

<img width="735" alt="screen shot 2018-02-05 at 1 54 31 pm" src="https://user-images.githubusercontent.com/627123/35823060-42f830ec-0a7c-11e8-8dd6-873493fdb04b.png">

the DOM APIs are returning a value of `''`, I'm open to rewording recommendations to make this more explicit.

Reply:
The value is fine, but the test says "sets data-shared-timefilter-* to null", but it's not null, and it's not testing for null. My only beef here is the description of what is being tested.

Reply:
Does `doesn't set data-shared-timefilter-* using the timefilter when auto-refresh selector is enabled` seem any more clear?

Reply:
That's fine. My only issue was the mention of `null`, since null is never the value and was not actually be tested for.